### ISSUE TITLE
why the labels in sampled_softmax_loss func are all zero?

### ISSUE THREAD

**Issue by universewill** [Stonesjtu/Pytorch-NCE#15]
I read the code in nce_loss.py:

```
    def sampled_softmax_loss(self, logit_target_in_model, logit_noise_in_model, logit_noise_in_noise, logit_target_in_noise):
        """Compute the sampled softmax loss based on the tensorflow's impl"""
        logits = torch.cat([logit_target_in_model.unsqueeze(2), logit_noise_in_model], dim=2)
        q_logits = torch.cat([logit_target_in_noise.unsqueeze(2), logit_noise_in_noise], dim=2)
        # subtract Q for correction of biased sampling
        logits = logits - q_logits
        labels = torch.zeros_like(logits.narrow(2, 0, 1)).squeeze(2).long()
        loss = self.ce(
            logits.view(-1, logits.size(-1)),
            labels.view(-1),
        ).view_as(labels)

        return loss
```

The labels are created by 'torch.zeros_like' function, so they are all zeros.
Is this a bug? Because the target label should be one?

**Comment by universewill** [comment]
I think zero label denotes the index of target.
